src/main/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJob.java

```diff
@@ -33,23 +37,50 @@ public HttpCallJob(
     }
 
     @Override
-    public void execute(JobExecutionContext context) {
+    public void execute(JobExecutionContext context) throws JobExecutionException {
         String jobId = context.getJobDetail().getKey().getName();
-        logger.info("Executing job " + jobId);
 
-        HttpAction action = actionExtractor.extract(context)
-            .withHeader("ServiceAuthorization", tokenGenerator.generate());
+        try {
+            logger.info("Executing job " + jobId);
 
-        ResponseEntity<String> response =
-            restTemplate
-                .exchange(
-                    action.url,
-                    action.method,
-                    toHttpEntity(action),
-                    String.class
-                );
+            HttpAction action = actionExtractor.extract(context)
+                .withHeader("ServiceAuthorization", tokenGenerator.generate());
 
-        logger.info("Job {} executed. Response code: {}", jobId, response.getStatusCodeValue());
+            ResponseEntity<String> response =
+                restTemplate
+                    .exchange(
+                        action.url,
+                        action.method,
+                        toHttpEntity(action),
+                        String.class
+                    );
+
+            logger.info("Job {} executed. Response code: {}", jobId, response.getStatusCodeValue());
+
+            failIfErrorResponse(response, jobId);
+        } catch (JobExecutionException e) {
+            throw e;
+        } catch (Exception e) {
+            String errorMessage = String.format("Job failed. Job ID: %s", jobId);
+            logger.error(errorMessage, e);
+            throw new JobExecutionException(errorMessage, e);
+        }
+    }
+
+    private void failIfErrorResponse(
+        ResponseEntity<?> response,
+        String jobId
+    ) throws JobExecutionException {
+
+        if (response.getStatusCodeValue() >= 400) {
+            throw new JobExecutionException(
+                String.format(
+                    "Job failed - received an error HTTP response (%s). Job ID: %s.",
+                    response.getStatusCodeValue(),
+                    jobId
+                )
+            );
```

Comment:
since we don't pass extra exception, could just use built in constructor avoiding our own string message build